src-plugins/itkFilters/itkFiltersComponentSizeThresholdProcess.cpp

```diff
@@ -143,12 +147,12 @@ template <class PixelType> int itkFiltersComponentSizeThresholdProcess::castToUI
     typedef itk::CastImageFilter< InputImageType, OutputImageType > CastFilterType;
 
     typename CastFilterType::Pointer  caster = CastFilterType::New();
-    typename InputImageType::Pointer im = dynamic_cast< InputImageType*>((itk::Object*)(inputData->data()));
+    typename InputImageType::Pointer im = dynamic_cast< InputImageType*>((itk::Object*)(getInputData()->data()));
     caster->SetInput(im);
     caster->Update();
 
-    inputData = medAbstractDataFactory::instance()->createSmartPointer ( "itkDataImageUInt3" );
-    inputData->setData(caster->GetOutput());
+    setInputData(medAbstractDataFactory::instance()->createSmartPointer ( "itkDataImageUInt3" ));
```

Comment:
is this ok? This seems odd to erase completely the input

Reply:
But this is what was done before, only through direct access to inputData instead of through a setter function.

Reply:
ok i looked again the code, and it's in `itkFiltersComponentSizeThresholdProcess::castToUInt3`.
What could be better is to return the casted output from castToUInt3 and use it in `itkFiltersComponentSizeThresholdProcess::updateProcess` directly. In that case, we won't need this setInputData() function which change the inputData (it could affect what the user wants if he/she uses an other process after this one).

Reply:
This only affects the internal state of this process. It is invisible from outside. The original image is not affected, so the user can use any other process, it won't change anything.

Reply:
(again, there is no functional change here, just the syntax change of using a setter instead of direct access to a variable)

Reply:
huuuum, don't like too much too, but why not...